tests/migration/patch_system.pm

```diff
@@ -0,0 +1,70 @@
+# SLE15 migration tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Fully patch the system and install patterns before conducting migration
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Utils::Architectures;
+use migration;
+use version_utils qw(is_desktop_installed is_upgrade is_sles4sap);
+use Utils::Backends 'is_pvm';
+use y2_base;
+
+sub run {
+    select_console 'root-console';
+
+    my ($self) = @_;
+
+    # Do not attempt to log into the desktop of a system installed with SLES4SAP
+    # being prepared for upgrade, as it does not have an unprivileged user to test
+    # with other than the SAP Administrator
+    my $nologin = (get_var('HDDVERSION') and is_upgrade() and is_sles4sap());
+
+    if (get_var('MINIMAL_UPDATE')) {
+        minimal_patch_system();
+    }
+    else {
+        fully_patch_system();
+        # Sometimes update package 'polkit' will cause GDM restart, so after
+        # update patches we'd better to select_console to make test robust.
+        select_console 'root-console';
+        # Perform sync ahead of reboot to flush filesystem buffers
+        assert_script_run 'sync', 600;
+        # Open gdm debug info for poo#45236, this issue happen sometimes in openqa env
+        script_run('sed -i s/#Enable=true/Enable=true/g /etc/gdm/custom.conf');
+        # Remove '-f' for reboot for poo#65226
+        enter_cmd "reboot";
+        reconnect_mgmt_console if is_pvm;
+        $self->wait_boot(textmode => !is_desktop_installed(), ready_time => 600, bootloader_time => 300, nologin => $nologin);
+        # Setup again after reboot
+        $self->setup_sle();
+    }
+
+    select_console 'root-console';
+    install_patterns() if (get_var('PATTERNS'));
+
+    # Record the installed rpm list
+    assert_script_run 'rpm -qa > /tmp/rpm-qa.txt';
+    upload_logs '/tmp/rpm-qa.txt';
+
+    # mark system patched
```

Comment:
from this line to the end of the function we could remove it, the registration I guess we can control by schedule, but I'm not sure how it affect the rest, and that the system is patched...I don't get that part why is needed to be set to 1.